packages/react-router-native/package.json

```diff
@@ -26,6 +27,8 @@
     "babel-preset-react-native": "1.9.1",
     "jest": "18.1.0",
     "react": "15.4.1",
+    "react-addons-test-utils": "^15.4.2",
+    "react-dom": "^15.4.2",
```

Comment:
`react-dom` is required for `react-addons-test-utils`. I opted for this because it provides an easy way to shallowly render components that require context. Shallow renders provide direct access to the component making them easier to unit test.

Here's the output compared to that of react-test-renderer. I tried to find the rendered TouchableHighlight using react-test-renderer, but I couldn't seem to locate it to unit test it. I felt like the last option provided the least amount of noise while still managing to hit all of the important assertions. I'm definitely open to suggestions though.

---

Using `react-test-renderer`
```jsx
import renderer from 'react-test-renderer'
const output = renderer.create(<NativeRouter><Link to='/'><View /></Link></NativeRouter>)
```

<details>
<summary>output</summary><p>

```jsx
// nothing useful
{
  type: 'View',
  props: {
    accessible: true,
    accessibilityLabel: undefined,
    accessibilityComponentType: undefined,
    accessibilityTraits: undefined,
    style: [
      [Object], undefined
    ],
    onLayout: undefined,
    hitSlop: undefined,
    isTVSelectable: true,
    tvParallaxProperties: undefined,
    hasTVPreferredFocus: undefined,
    onStartShouldSetResponder: {
      [Function: bound touchableHandleStartShouldSetResponder]
      __reactBoundContext: [Object],
      __reactBoundMethod: [Function: touchableHandleStartShouldSetResponder],
      __reactBoundArguments: null,
      bind: [Function]
    },
    onResponderTerminationRequest: {
      [Function: bound touchableHandleResponderTerminationRequest]
      __reactBoundContext: [Object],
      __reactBoundMethod: [Function: touchableHandleResponderTerminationRequest],
      __reactBoundArguments: null,
      bind: [Function]
    },
    onResponderGrant: {
      [Function: bound touchableHandleResponderGrant]
      __reactBoundContext: [Object],
      __reactBoundMethod: [Function: touchableHandleResponderGrant],
      __reactBoundArguments: null,
      bind: [Function]
    },
    onResponderMove: {
      [Function: bound touchableHandleResponderMove]
      __reactBoundContext: [Object],
      __reactBoundMethod: [Function: touchableHandleResponderMove],
      __reactBoundArguments: null,
      bind: [Function]
    },
    onResponderRelease: {
      [Function: bound touchableHandleResponderRelease]
      __reactBoundContext: [Object],
      __reactBoundMethod: [Function: touchableHandleResponderRelease],
      __reactBoundArguments: null,
      bind: [Function]
    },
    onResponderTerminate: {
      [Function: bound touchableHandleResponderTerminate]
      __reactBoundContext: [Object],
      __reactBoundMethod: [Function: touchableHandleResponderTerminate],
      __reactBoundArguments: null,
      bind: [Function]
    },
    testID: undefined
  },
  children: [{
    type: 'Text',
    props: [Object],
    children: null
  }]
}
```

</p></details>

---
Digging into `react-test-renderer`
```jsx
import renderer from 'react-test-renderer'
const wrapper = renderer.create(<NativeRouter><Link to='/'><View /></Link></NativeRouter>)
const ouput = wrapper._component._currentElement.props.child.props.children
```

<details>
<summary>output</summary><p>

```jsx
// doesn't actually render the TouchableHighlight
{
  '$$typeof': Symbol(react.element),
  type: {
    [Function: Link]
    contextTypes: {
      router: [Function: bound checkType]
    },
    propTypes: {
      onPress: [Object],
      component: [Object],
      replace: [Object],
      to: [Object]
    },
    defaultProps: {
      component: [Object],
      replace: false
    }
  },
  key: null,
  ref: null,
  props: {
    to: '/',
    children: {
      '$$typeof': Symbol(react.element),
      type: [Function: Component],
      key: null,
      ref: null,
      props: [Object],
      _owner: null,
      _store: {}
    },
    component: {
      [Function]
      displayName: 'TouchableHighlight',
      propTypes: [Object],
      getDefaultProps: [Object],
      defaultProps: [Object]
    },
    replace: false
  },
  _owner: null,
  _store: {}
}
```

</p></details>

---

Using `react-addons-test-utils`
```jsx
import ReactTestUtils from 'react-addons-test-utils'
const renderer = ReactTestUtils.createRenderer()
renderer.render(<Link to='/' />, context)
const output = renderer.getRenderOutput()
```

<details>
<summary>output</summary><p>

```jsx
// easy to access with context
{
  '$$typeof': Symbol(react.element),
  type: {
    [Function]
    displayName: 'TouchableHighlight',
    propTypes: {
      accessible: [Object],
      accessibilityComponentType: [Object],
      accessibilityTraits: [Object],
      disabled: [Object],
      onPress: [Object],
      onPressIn: [Object],
      onPressOut: [Object],
      onLayout: [Object],
      onLongPress: [Object],
      delayPressIn: [Object],
      delayPressOut: [Object],
      delayLongPress: [Object],
      pressRetentionOffset: [Object],
      hitSlop: [Object],
      activeOpacity: [Object],
      underlayColor: [Object],
      style: [Function],
      onShowUnderlay: [Object],
      onHideUnderlay: [Object],
      hasTVPreferredFocus: [Object],
      tvParallaxProperties: [Object]
    },
    getDefaultProps: {
      [Function: getDefaultProps] isReactClassApproved: {}
    },
    defaultProps: {
      activeOpacity: 0.85,
      underlayColor: 'black'
    }
  },
  key: null,
  ref: null,
  props: {
    to: '/',
    replace: false,
    onPress: [Function],
    activeOpacity: 0.85,
    underlayColor: 'black'
  },
  _owner: null,
  _store: {}
}
```

</p></details>